jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ClientConnector.java

```diff
@@ -593,6 +602,7 @@ protected EndPoint newEndPoint(SelectableChannel channel, ManagedSelector select
         {
             EndPoint endPoint = ClientConnector.this.newEndPoint(channel, selector, selectionKey);
             endPoint.setIdleTimeout(getIdleTimeout().toMillis());
+            listener.onConnectSuccess((SocketChannel)channel);
```

Comment:
This line should be moved in the 2 places just after the connection has been established.

Reply:
Yeah, you are right!